### ISSUE TITLE
Add `Access-Control-Max-Age` header to response from OPTIONS request

### ISSUE THREAD

**Issue by andraaandrus** [apollographql/router#2535]
**Describe the solution you'd like**
When sending an OPTIONS request to the router we don't receive the `access-control-max-age` CORS header in the response:
```
curl -X OPTIONS -v http://localhost:4000/test/graphql
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000 (#0)
> OPTIONS /cornichon-0201-102605-422-0/graphql HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/7.86.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< access-control-allow-origin: *
< access-control-allow-methods: GET,POST,DELETE,OPTIONS
< access-control-allow-headers: accept,authorization,content-type,origin,user-agent,x-correlation-id
< content-length: 0
< date: Thu, 02 Feb 2023 10:27:39 GMT
< 
* Connection #0 to host localhost left intact
```

We would like to have the `Access-Control-Max-Age` header in the response from OPTIONS requests.

**Describe alternatives you've considered**
Tried adding the `max_age` cors config in router.yaml, as per [apollo docs](https://www.apollographql.com/docs/router/configuration/cors/#all-cors-options), but received error message:
```
    allow_any_origin: true
    methods: [ GET, POST, DELETE, OPTIONS ]
    allow_headers: [ Accept, Authorization, Content-Type, Origin, User-Agent, X-Correlation-ID ]
    expose_headers: [ X-Correlation-ID ]
┌   max_age: 2h
└-----> Additional properties are not allowed ('max_age' was unexpected)
```

We have also tried adding the header via plugin using the `router_service`. While this worked for GET requests, it does not work for OPTIONS requests.

```
curl -v http://localhost:4000/test/graphql
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000 (#0)
> GET /cornichon-0201-102605-422-0/graphql HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/7.86.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< content-type: application/json
< access-control-allow-headers: accept,authorization,content-type,origin,user-agent,x-correlation-id
< access-control-allow-methods: GET,POST,DELETE,OPTIONS
< access-control-max-age: 299
< content-length: 118
< access-control-allow-origin: *
< access-control-expose-headers: x-correlation-id
< date: Thu, 02 Feb 2023 10:27:04 GMT
< 
* Connection #0 to host localhost left intact
```

**Comment by thisisomar** [comment]
What version of Apollo Router are you on?

The ACCESS CONTROL MAX AGE header option is only available on the newest version (v1.10.0)

If you're on anything earlier, it won't be there

**Comment by andraaandrus** [comment]
Hi @thisisomar , we were running v0.9.0. Will upgrade and let you know if we still have issues. Thanks for the response!
